js/src/components/paid-ads/ads-campaign/ads-campaign.js

```diff
@@ -102,13 +103,11 @@ export default function AdsCampaign( {
 
 			<StepContentFooter>
 				<StepContentActions>
-					<AppButton
-						isPrimary
-						disabled={ ! isValidForm }
-						onClick={ onContinue }
-					>
-						{ __( 'Continue', 'google-listings-and-ads' ) }
-					</AppButton>
+					<ContinueButton
+						onContinue={ onContinue }
+						setupAdsFlow={ trackingContext === 'setup-ads' }
+						isLoading={ isLoading }
+					/>
```

Comment:
Thanks for the further update! However, it doesn't solve the concern in https://github.com/woocommerce/google-listings-and-ads/pull/2577#discussion_r1780557466.

To externalize the button, the changes may be like:
- In `AdsCampaign`
   - After externalization, `AdsCampaign` won't have the `loading` and `onContinue` props
   - The continue button is given via a prop or a render prop
   - For example
      ```jsx
      export default function AdsCampaign( {
      	campaign,
      	continueButton,
      	trackingContext,
      } ) {
      	// ... omitted
      
      	return (
      		<StepContent>
      			// ... omitted
      			<StepContentFooter>
      				<StepContentActions>
      					{ typeof continueButton === 'function'
      						? continueButton( formContext )
      						: continueButton }
      				</StepContentActions>
      				<PaidAdsFaqsPanel />
      			</StepContentFooter>
      		</StepContent>
      	);
      }
      ```
- In `AdsStepper`
   - The [ContinueButton](https://github.com/woocommerce/google-listings-and-ads/blob/fe7bab660d13bab427fa28a3b37e7f51569d5865/js/src/components/paid-ads/ads-campaign/continue-button.js) can be inlined and passed via the `continueButton` prop
   - For example
      ```jsx
      const AdsStepper = ( { formProps } ) => {
      	// ... omitted
      
      	const isDisabledLaunch = ! formProps.isValidForm; // and also the conditions in https://github.com/woocommerce/google-listings-and-ads/blob/fe7bab660d13bab427fa28a3b37e7f51569d5865/js/src/components/paid-ads/ads-campaign/continue-button.js#L34-L37
      
      	return (
      		<Stepper
      			// ... omitted
      			steps={ [
      				// ... omitted
      				{
      					key: '2',
      					label: __(
      						'Create your paid campaign',
      						'google-listings-and-ads'
      					),
      					content: (
      						<AdsCampaign
      							trackingContext="setup-ads"
      							continueButton={
      								<AppButton
      									isPrimary
      									text={ __(
      										'Launch paid campaign',
      										'google-listings-and-ads'
      									) }
      									disabled={ isDisabledLaunch }
      									loading={ formProps.isSubmitting }
      									onClick={ formProps.handleSubmit }
      								/>
      							}
      						/>
      					),
      					onClick: handleStepClick,
      				},
      			] }
      		/>
      	);
      };
      ```
- In `CreatePaidAdsCampaign` and `EditPaidAdsCampaign`
   - Similar to `AdsStepper`, change to pass the button via the `continueButton` prop
   - For example
      ```jsx
      const SetupAdsForm = () => {
      	// ... omitted
      
      	// This can be a component shared between CreatePaidAdsCampaign and EditPaidAdsCampaign
      	const ContinueButton = ( formProps ) => {
      		return (
      			<AppButton
      				isPrimary
      				text={ __( 'Continue', 'google-listings-and-ads' ) }
      				disabled={ ! formProps.isValidForm }
      				onClick={ () => handleContinueClick( STEP.ASSET_GROUP ) }
      			/>
      		);
      	};
      
      	return (
      		<>
      			// ... omitted
      			<CampaignAssetsForm
      				initialCampaign={ {
      					amount: 0,
      					countryCodes: initialCountryCodes,
      				} }
      				onSubmit={ handleSubmit }
      			>
      				<Stepper
      					currentStep={ step }
      					steps={ [
      						{
      							key: STEP.CAMPAIGN,
      							label: __(
      								'Create paid campaign',
      								'google-listings-and-ads'
      							),
      							content: (
      								<AdsCampaign
      									trackingContext={ eventContext }
      									continueButton={ ContinueButton }
      								/>
      							),
      							onClick: handleStepperClick,
      						},
      						// ... omitted
      					] }
      				/>
      			</CampaignAssetsForm>
      		</>
      	);
      };
      ```

In this way, the logic of this button will be split up at a higher level for each consumer component, further avoiding the need to add/pass on different contextual conditions to the `AdsCampaign` and centralize all the conditions with the same button.